src/Api/SecretsManager/Controllers/SecretsController.cs

```diff
@@ -246,4 +248,38 @@ public async Task<ListResponseModel<BulkDeleteResponseModel>> BulkDeleteAsync([F
         var responses = secrets.Select(s => new BaseSecretResponseModel(s));
         return new ListResponseModel<BaseSecretResponseModel>(responses);
     }
+
+    [HttpPost("organizations/{organizationId}/secrets/move")]
+    public async Task BulkMoveToProjectAsync(Guid organizationId, [FromBody] BulkMoveToProjectsRequestModel request)
+    {
+        var secrets = (await _secretRepository.GetManyByIds(request.Secrets)).ToList();
+        if (!secrets.Any() || secrets.Count != request.Secrets.Count)
+        {
+            throw new NotFoundException();
+        }
+
+        if (secrets.Any(secret => secret.OrganizationId != organizationId) ||
+            !_currentContext.AccessSecretsManager(organizationId))
```

Comment:
It looks like we are already checking `!_currentContext.AccessSecretsManager(organizationId)` in 
` _authorizationService.AuthorizeAsync(
            User, secrets.AsReadOnly(), BulkSecretOperations.Update);`
            
 Should we remove it here since it is an authz concern?